### DISCUSSION TITLE
How is a function called

### DISCUSSION THREAD

**Discussion by candcconsulting** [blib-la/hyv#13] [Q&A]
Hi, this looks like a great project and certainly like the way that the reasoner works.
However, I am struggling to get the functions to work
I am using the llamacpp server and it is responding ... 
I get an output message
{
  '{thought:TheUserIsAskingForTheWeatherAndPopulationOfBerlin,GermanyIWillUseTheGetWeatherAndGetPopulationFunctionsToRetrieveThisInformation,reason:ByFollowingTheInstructionsGivenInTheTemplateAndUsingTheDefinedFunctions,ICanProvideAnAccurateAnswer,reflection:ThisTaskRequiresMeToApplyMyProblemSolvingAbilitiesByUtilizingTheProvidedFunctionsEffectively,answer:{': '"answers":[\n' +
    '{\n' +
    '"weather": {\n' +
    '"result": "{}",\n' +
    '"error": null,\n' +
    '"functionCalls": [{\n' +
    '"functionName": "getWeather",\n' +
    '"arguments": {"location": "Berlin, Germany"}\n' +
    '}]\n' +
    '},\n' +
    '"population": {\n' +
    '"result": "{}",\n' +
    '"error": null,\n' +
    '"functionCalls": [{\n' +
    '"functionName": "getPopulation",\n' +
    '"arguments": {"location": "Berlin, Germany"}\n' +
    '}]\n' +
    '}\n' +
    '}\n' +
    ']\n' +
    '}}'
}
but the function is never called ... 
OK the json is a bit messed up ... but I am handling that
Also it would seem to get the function calling I had to inject the functions into the system prompt .. I will do some more digging ... but I think  am missing something obvious.

As you can see I am using the simple example from your examples folder

**Comment by pixelass** [comment]
Sorry for the late reply. We totally missed the comment. Sadly Hyv never gained any traction so we moved on. Functions and other newer API was never implemented. We might revisit this project but right now we're working on other groundbreaking ideas. We highly recommend looking at langchain or the AI tools provided by Vercel.
